src/components/TodoFooter.tsx

```diff
@@ -0,0 +1,82 @@
+import { Todo } from '../types/Todo';
+import { getActiveTodos, getCompletedTodos, getTodos } from '../api/todos';
+import { useState } from 'react';
+import classNames from 'classnames';
+import { Filter } from '../types/Filter';
+
+type Props = {
+  todos: Todo[];
+  filterTodos: (filteredTodos: Todo[]) => void;
+};
+
+export const TodoFooter: React.FC<Props> = ({
+  todos,
+  filterTodos = () => {},
+}) => {
+  const [filterType, setFilterType] = useState<Filter>(Filter.All);
+
+  const todosCount = (todosType: Filter): number => {
+    const value = todosType === Filter.Active ? false : true;
+
+    return todos.filter(todo => todo.completed === value).length;
+  };
+
+  const filterFunction = (filter: Filter) => {
+    setFilterType(filter);
+
+    switch (filter) {
+      case Filter.All:
+        getTodos().then(filterTodos);
+        break;
+
+      case Filter.Active:
+        getActiveTodos().then(filterTodos);
+        break;
+
+      case Filter.Completed:
+        getCompletedTodos().then(filterTodos);
+        break;
+
+      default:
+        getTodos().then(filterTodos);
+    }
+  };
+
+  //eslint-disable-next-line
+  function findFilterKey(enumObj: any, value: string): string | undefined {
```

Comment:
It is important to follow the same way of creating functions everywhere and don't use `any` as type
```suggestion
  const findFilterKey = (enumObj: any, value: string): string | undefined => {
```